src/aioftp/client.py

```diff
@@ -99,29 +168,55 @@ async def finish(self, expected_codes="2xx", wait_codes="1xx"):
         self.close()
         await self.client.command(None, expected_codes, wait_codes)
 
-    async def __aexit__(self, exc_type, exc, tb):
+    @override
+    async def __aexit__(
+        self,
+        exc_type: Optional[Type[BaseException]],
+        exc: Optional[BaseException],
+        tb: Optional[TracebackType],
+    ) -> None:
         if exc is None:
             await self.finish()
         else:
             self.close()
 
 
+CodesORCode: TypeAlias = Union[Tuple[Code, ...], Code]
+EmptyCodes: TypeAlias = Tuple[()]
+
+
 class BaseClient:
+    socket_timeout: Optional[int]
+    connection_timeout: Optional[int]
+    throttle: StreamThrottle
+    path_timeout: Optional[int]
+
+    path_io: pathio.AbstractPathIO
+    encoding: str
+    stream: Optional[ThrottleStreamIO]
+    ssl: Optional[SSLContext]
+    parse_list_line_custom: Optional[Callable[[bytes], Tuple[pathlib.PurePosixPath, InfoDict]]]
+    parse_list_line_custom_first: bool
+    _passive_commands: Sequence[str]
+    _open_connection: "partial[Coroutine[None, None, Tuple[asyncio.StreamReader, asyncio.StreamWriter]]]"
+    server_host: Optional[str] = None
+    server_port: Optional[int] = None
```

Comment:
This describes what types the variables available in the object of this class have. Dataclass or Pydantic have nothing to do with it. It's just like in c++, for example. As it seems to me, there are two ways. The stricter one, as in this code, when all available variables are labeled by type through self. Or there is also a more non-strict way when variables are marked only in constructor arguments. But in this case there may still be some variables that are not initialized in the constructor and it will also need to be described in the class body. Besides, this variant leaves a little more room for errors when the class implies the use of one type, but the constructor uses another type.  That's why I prefer to describe everything explicitly in the class body. But if it seems unnecessary I can of course remove it.

Reply:
https://peps.python.org/pep-0526/#class-and-instance-variable-annotations

Reply:
There is indeed a difference in your example. It is very important to distinguish when a variable has initialization at once (then it is both a class object variable and a class variable). But in my code it is written in most cases:
```python
class A:
    x: int
```
This is exactly and means that the class object has a variable of this type.

Reply:
Look at the pep examples I linked above, they use exactly the same style of annotation of class object variables as mine.

Reply:
Okay. Here is a simplified example that exists in the project:

```python
class ThrottleStreamIO:
    pass


class BaseClient:
    def __init__(self) -> None:
        self.stream = None

    async def connect(self, host: str) -> None:
        self.server_host = host
        self.stream = ThrottleStreamIO()
```

Here mypy reports on the line `self.stream = ThrottleStreamIO()` - `Incompatible types in assignment (expression has type "ThrottleStreamIO", variable has type "None")`. In addition, on the line `self.server_host = host` pycharm reports `Instance attribute server_host defined outside __init__ ` This is exactly solved by adding an annotation to the class body. If you don't like more explicitly stating the types of variables available in the class, I can remove annotations for most declarations (except in the cases above).

Reply:
The code from the project is pretty old and I also suggest to refactor such cases to
```python
class BaseClient:
    def __init__(self) -> None:
        self.server_host: str | None = None
        self.stream: ThrottleStreamIO | None = None

    async def connect(self, host: str) -> None:
        self.server_host = host
        self.stream = ThrottleStreamIO()
```
Or even
```python
class Client:
    def __init__(self, server_host: str, stream: ThrottleStreamIO):
        self.server_host = server_host
        self.stream = stream

async def connect(host: str) -> Client:
    stream = ThrottleStreamIO()
    client = Client(host, stream)
    return client
```
to reduce unions.

Reply:
Done. Removed unnecessary annotations, only left in a few places where they are really needed